microprofile/WFLY-12681_Provide_a_microprofile-platform_Galleon_layer.adoc

```diff
@@ -0,0 +1,86 @@
+= WFLY-12681 Provide a 'microprofile-platform' Galleon layer
+:author:            Radoslav Husar
+:email:             rhusar@redhat.com
+:toc:               left
+:icons:             font
+:idprefix:
+:idseparator:       -
+
+== Overview
+
+Historically, we used to have a 'microprofile' layer spec aggregating all MicroProfile (further as MP) specs.
+At that moment it only contained config, health and metrics which got eventually renamed
+to 'observability' by https://issues.redhat.com/browse/WFLY-11774[WFLY-11774] (no RFE there).
+
+With the introduction of new the specs such as MP Fault Tolerance, MP OpenAPI, etc. we want to
+reintroduce the 'microprofile-platform' layer spec which contains all available MP specs in the server.
+
+== Issue Metadata
+
+=== Issue
+
+* https://issues.redhat.com/browse/WFLY-12681[WFLY-12681 Provide a 'microprofile-platform' Galleon layer]
+* https://issues.redhat.com/browse/EAP7-1396[EAP7-1396 Provide a 'microprofile-platform' Galleon layer]
+
+=== Related Issues
+
+* https://issues.redhat.com/browse/WFLY-11774[WFLY-11774 Rename microprofile layer to be observability]
+
+=== Dev Contacts
+
+* mailto:rhusar@redhat.com[Radoslav Husar]
+
+=== QE Contacts
+
+* TBD
+
+=== Testing By
+
+* [x] Engineering
+
+* [ ] QE
+
+=== Affected Projects or Components
+
+* MicroProfile
+* Galleon
+
+=== Other Interested Projects
+
+N/A
+
+== Requirements
+
+=== Hard Requirements
+
+* New 'microprofile-platform' Galleon layer is available for provisioning.
+* Each MP spec is optional to allow for excluding during provisioning (this is a community-only feature).
+
+=== Nice-to-Have Requirements
+
+N/A
+
+=== Non-Requirements
+
+N/A
+
+== Implementation Plan
+
+No coordination required. Only thing that needs to make use of this are quickstart examples using bootable jars should
+make use of the new layer in the product.
+
+== Test Plan
+
+Testing is encompassed in the upstream testsuite in multiple places:
+
+* by the `testsuite/integration/basic` which provisions this layer and runs subset of MP tests from basic suite with it
+* by the `testsuite/integration/microprofile` test module which contains MP integration tests
+* by the `testsuite/layers` test module for testing modules
+
```

Comment:
@rhusar given this belongs to a series of related RFEs why don't we take inspiration from the [JSF RFE](https://github.com/wildfly/wildfly-proposals/pull/310/) and re-factor the paragraph into something like:

The test coverage of the Galleon layer added by this proposal is divided in two main groups:

Testing the Galleon layer provisioning. This testing is done by https://github.com/wildfly/wildfly/blob/master/testsuite/layers/src/test/java/org/jboss/as/test/layers/LayersTestCase.java[LayersTestCase]. The testsuite will be modified to add a new server provisioned with this layer in isolation and with this layer combined with all the layers. For each kind of provisioning, this test does the following:

.. Verifies the provisioned modules are the expected ones.
.. Verifies the provisioned server starts successfully.

. Execution of WildFly tests related to the feature provisioned by this layer. Reuse the existing tests available on the WildFly test suite, which are directly testing this layer functionalities, and execute them on a server installation provisioned with this layer:

.. by the `testsuite/integration/basic` which provisions this layer and runs subset of MP tests from basic suite with it
.. by the `testsuite/integration/microprofile` test module which contains MP integration tests (all existing tests)

Reply:
@tommaso-borgato Ok, used your text even though I liked mine better - since it was simple and clear :-)

Reply:
@rhusar thank you